python/pdo/service_client/servicedatabase.py

```diff
@@ -0,0 +1,292 @@
+            self.file_name = file_name
+            try:
+                self.load_data_from_file()
+            except Exception as e:
+                logger.exception('Failed to load database during initialization' + str(e))
+                raise Exception from e
+        else:
+            self.data = dict()
+
+        self._service_type = service_type
+        
+        if (self._service_type is not 'eservice') and (self._service_type is not 'sservice'):
+            logger.exception('Cannot initialize service database manager: service_type must be eservice or sservice')
+            raise Exception('Cannot initialize service database manager: service_type must be eservice or sservice')
+
+
+    #--------------------------------------------
+    #--------------------------------------------
+
+    @staticmethod
+    def get_eservice_client(url):
+        try :
+            client = EnclaveServiceClient(url)
+        except Exception as e :
+            logger.exception('failed to contact enclave service; %s', str(e))
+            raise Exception from e
+        return client
+
+    #--------------------------------------------
+    #--------------------------------------------
+
+    @staticmethod
+    def get_sservice_client(url):
+        try :
+            client = StorageServiceClient(url)
+        except Exception as e :
+            logger.exception('failed to contact storage service; %s', str(e))
+            raise Exception from e
+        return client
+
+    def load_data_from_file(self):
+        """ Load the json data file as a dictionary"""
+
+        if os.path.exists(self.file_name):
+            try:
+                with open(self.file_name, 'r') as fp:
+                    self.data = json.load(fp)
+                    if not isinstance(self.data, dict):
+                        raise Exception('Invalid json file for service database: Json file must be loadable as python dictionary')
+            except Exception as e:
+                logger.exception('Failed to load json file for service database: ' + str(e))
+                raise Exception from e
+        else:
+            raise Exception('Cannot load service database: Data file does not exist')
+    
+    #--------------------------------------------
+    #--------------------------------------------
+    
+    def save_data_to_file(self, new_file_name = None):
+        """ Save the dictionary as a json file. If no new_file_name is provided, the json file used for init will be overwritten"""
+        if new_file_name is not None:
+            self.file_name = new_file_name
+
+        if self.file_name is None:
+            logger.exception('Cannot save service database info without providing json file name')
+            raise Exception('Cannot save service database info without providing json file name')
+        
+        try:
+            with open(self.file_name, 'w') as fp:
+                json.dump(self.data, fp)
+        except Exception as e:
```

Comment:
only issue with this is that when the write fails you may corrupt the old file. write to a temp & mv to the actual.

Reply:
oops. thank you for pointing out. will fix this.